idaes/generic_models/unit_models/heat_exchanger.py

```diff
@@ -162,19 +163,20 @@ def delta_temperature_underwood_callback(b):
     This uses a cube root function that works with negative numbers returning
     the real negative root. This should always evaluate successfully.
     """
+    dT1 = b.delta_temperature_in
+    dT2 = b.delta_temperature_out
+    temp_units = pyunits.get_units(dT1)
+
     # external function that ruturns the real root, for the cuberoot of negitive
     # numbers, so it will return without error for positive and negitive dT.
     b.cbrt = ExternalFunction(
```

Comment:
The output of this should probably have units too; i.e. `units=pyunits.K`. This would mean you don't need to add units on line 179.

Reply:
I think this is ok.  I just left the cube root function return value unitless.  I figured this would make things simpler than having units of cube root of K the get cubed back to K later.  That process made me a little uncomfortable. Like what if I ended up with units of K^0.9999999999999 or something?

Reply:
@jsiirola or @carldlaird, if we have units like (K^(1/3))^3 are we sure that the resulting unit is exactly K and not just close to K?